pyrefinebio/script.py

```diff
@@ -0,0 +1,126 @@
+import click
+import json
+
+import pyrefinebio.high_level_functions as hlf 
+
+from pyrefinebio.exceptions import DownloadError
+
+@click.group()
+def cli():
+    """RefineBio CLI
+    """
+    pass
+
+class DictParamType(click.ParamType):
+    name = "dict"
+
+    def convert(self, value, param, ctx):
+        try:
+            return json.loads(value)
+        except:
+            self.fail(
+                "expected valid string in a dict form, "
+                "got: {0}".format(value),
+                param,
+                ctx
+            )
+
+
+class ListParamType(click.ParamType):
+    name = "list"
+
+    def convert(self, value, param, ctx):
+        try:
+            return value.split()
+        except:
+            self.fail(
+                "expected valid string in list form, "
+                "got: {0}".format(value),
+                param,
+                ctx
+            )
+
+
+@cli.command()
+@click.option("--entity", default="", help="entity to print help for")
+def help(entity=None):
+    """
+    Prints out information about pyrefinebio's classes and functions.
+    To get information about a class method, pass in the class name and method name  
+    separated by either a space or a `.`
+
+    Example:
+        $ refinebio help --entity "Sample.search"
+    """
+    hlf.help(entity)
+
+
+@cli.command()
+@click.option("--path", required=True, help="Path that the dataset should be downloaded to")
+@click.option("--email-address", required=True, help="The email that will be contacted with info related to the Dataset")
+@click.option("--dataset-dict", default=None, type=DictParamType(), help="A fully formed Dataset `data` attribute. Use this parameter if you want to specify specific Samples for your Dataset. Ex: '{\"GSE74410\": [\"GSM1919903\"]}' ")
+@click.option("--experiments", default=None, type=ListParamType(), help="A space separated list of experiment accession codes. Ex: 'SRP051449 GSE44421 GSE44422'")
+@click.option("--aggregation", default="EXPERIMENT", type=click.Choice(("EXPERIMENT", "SPECIES", "ALL"), case_sensitive=True), help="How the Dataset should be aggregated")
+@click.option("--transformation", default="NONE", type=click.Choice(("NONE", "MINMAX", "STANDARD"), case_sensitive=True), help="The transformation for the Dataset")
+@click.option("--skip-quantile-normalization", default=False, help="Control whether the Dataset should skip quantile normalization for RNA-seq Samples")
```

Comment:
These lines are too long, they're kind of hard to read. [PEP8](https://www.python.org/dev/peps/pep-0008/) says:

```
The default wrapping in most tools disrupts the visual structure of the code, making it more difficult to understand. The limits are chosen to avoid wrapping in editors with the window width set to 80, even if the tool places a marker glyph in the final column when wrapping lines. Some web based tools may not offer dynamic line wrapping at all.

Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the line length limit up to 99 characters, provided that comments and docstrings are still wrapped at 72 characters.
```

I fall into the "prefer a longer line length" of 100. If reformatting these to be shorter seems tedious, you could just run [black](https://github.com/psf/black) on them with `black --line-length=100 .`